jetty-websocket/websocket-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketUpgradeRequest.java

```diff
@@ -582,7 +582,11 @@ public void upgrade(HttpResponse response, HttpConnectionOverHTTP oldConn)
 
         if (connectionListeners != null)
         {
-            connectionListeners.forEach((listener) -> connection.addListener(listener));
+            connectionListeners.forEach((listener) ->
+            {
+                if (!(listener instanceof WebSocketSession))
```

Comment:
I can see that this fixes the problem... but it is rather ugly and special case.
It may be better to not have the sessions directly as beans, but rather have a collection of sessions as the bean?

Reply:
Should probably backport the `SessionTracker` from d8a9f2619bb773a5d8d821ee5831950c1b58ba67 (now in `jetty-10.0.x`)